src/Sushi.Mediakiwi.API/UrlResolver.cs

```diff
@@ -548,7 +548,18 @@ public async Task ResolveUrlAsync(string uriScheme, HostString uriHost, PathStri
             if (string.IsNullOrWhiteSpace(targetName) == false)
             {
                 var urldecrypt = Utils.FromUrl(targetName);
-                var list = await Data.ComponentList.SelectOneAsync(urldecrypt, null).ConfigureAwait(false);
+                var lists = await Data.ComponentList.SelectAllAsync(urldecrypt, null).ConfigureAwait(false);
+                Data.IComponentList list = null;
+
+                if (lists?.Length > 1 && Folder?.ID > 0)
+                {
+                    list = lists.First(x => x.FolderID == Folder.ID);
+                }
+                else 
+                {
+                    list = lists.First();
+                }
```

Comment:
Could the list be empty? In that case .First() will throw an exception.

Reply:
Yep, fixed